Client/Frontend/Browser/TabManager.swift

```diff
@@ -173,6 +173,11 @@ class TabManager: NSObject {
             return
         }
 
+        // Make sure to wipe the private tabs if the user has the pref turned on
+        if shouldClearPrivateTabs(), !(tab?.isPrivate ?? false) {
+            removeAllPrivateTabsAndNotify(false)
```

Comment:
the previous tab is nil in a lot of cases. Most of the time when we call selectTab we never pass anything through. 
 So using it to check if the previous tab was a private tab doesnt work the way you want. Instead as long as the tab we are switching to is _not_ private we can be sure that removing any private tabs would be okay.  If we are switching from one normal tab to the other then thats totally fine if we remove any private tabs (there shouldnt be any).